### PR TITLE
chore: remove unused UPE JS methods

### PR DESCRIPTION
**Pull request by frosso**
Fixes #

#### Changes proposed in this Pull Request

I noticed that some of these functions aren't used anywhere. Removing.

#### Testing instructions

*

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/checkout/api/index.js [hunk 1]
```diff
@@ -117,90 +117,6 @@ export default class WCPayAPI {
...
 	}
 
-	/**
-	 * Generates a Stripe payment method.
-	 *
-	 * @param {Object} elements A hash of all Stripe elements, used to enter card data.
-	 * @param {Object} preparedCustomerData Default values for customer data, used on pages like Pay for Order.
-	 * @return {Object} A request object, which will be prepared and then `.send()`.
-	 */
-	generatePaymentMethodRequest( elements, preparedCustomerData = {} ) {
```
Comment (frosso) [inline client/checkout/api/index.js:127]: Not called anywhere in the codebase - search for `generatePaymentMethodRequest`

File: client/checkout/blocks/generate-payment-method.js [hunk 2]
```diff
@@ -1,63 +0,0 @@
...
- * Handles the payment method generation action.
- *
- * @param {WCPayAPI} api The API class that is used to connect both with the server and Stripe.
- * @param {Object} elements A hash, containing all Stripe card elements.
- * @param {Object} billingData The billing data, which was collected from the checkout block.
- * @param {string} fingerprint User fingerprint.
- *
- * @return {Object} The `onPaymentSetup` response object, including a type and meta data/error message.
- */
-const generatePaymentMethod = async (
```
Comment (frosso) [inline client/checkout/blocks/generate-payment-method.js:16]: Not used anywhere in the codebase

File: client/checkout/blocks/payment-elements.js [hunk 3]
```diff
@@ -60,42 +60,40 @@ const PaymentElements = ( { api, ...props } ) => {
 	] );
 
 	return (
-		<>
```
Comment (frosso) [inline client/checkout/blocks/payment-elements.js:63]: Removed the useless fragment I added by mistake

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8843` or branch name `chore/remove-unused-upe-js-methods` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=chore/remove-unused-upe-js-methods) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 7fdb13a96ed979844543d9efe080f7e188c9539f
- Build time: 2024-05-18 20:50:08 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by gpressutto5 — APPROVED** [review]

**Pull request merged**